### DISCUSSION TITLE
Functions and data types initialized in kotlin available for use in Android and IOS native code?

### DISCUSSION THREAD

**Discussion by lemkoleg** [buijs-dev/klutter#13] [General]
Hello. Are the functions and data types initialized in kotlin available for use in Android and IOS native code?

**Comment by buijs-dev** [comment]
Hey! I am not sure if I understand you're question... 

If you want to use the code that's written in the root/platform module, then yes that's available for use in a native app. Add the .aar file and/or xcframework to your native project. 

Not sure why you'd want to use klutter for that though? A simple KMP library will suffice.

**Reply by lemkoleg** [reply]
I have a kotlin project. It has a class responsible for receiving files, storing, and so on. I want to override a data source for native media players targeting this class. And then use betterplayer, for the front-end flutter project, which has a setting for those native players. I have two options: 1. generate native libraries for Android and IOS, then connect them to native projects. Then, all user classes and data types initialized in Kotlin will be present separately both in the library for android and in IOS. And, in order to use them in flutter, they will need to be described in a common flutter class / data type. 2. Maybe your plugin generates native libraries from Kotlin code and describes them in flutter automatically. I describe correctly?

**Reply by lemkoleg** [reply]
That is, I mean, are Kotlin classes / data types available for use in the Klutter project, in the ios and android folders of this project.

**Reply by buijs-dev** [reply]
> I have a kotlin project. It has a class responsible for receiving files, storing, and so on. I want to override a data source for native media players targeting this class. And then use betterplayer, for the front-end flutter project, which has a setting for those native players. I have two options: 1. generate native libraries for Android and IOS, then connect them to native projects. Then, all user classes and data types initialized in Kotlin will be present separately both in the library for android and in IOS. And, in order to use them in flutter, they will need to be described in a common flutter class / data type. 2. Maybe your plugin generates native libraries from Kotlin code and describes them in flutter automatically. I describe correctly?

Okay so for this I would recommend to create a new klutter project and then use it's generated flutter plugin. I am not sure if you're in a hurry but it would be best to wait a few weeks until the next release is done (some breaking changes).

How to do it:
- Create a new klutter project (use Intellij plugin or cli tool)
- Move your kotlin code to the root/platform module. 
- Run Gradle build which creates the flutter plugin.

Now you can use your kotlin code from a flutter app as it is a flutter plugin. I don't know how much code you have in kotlin, but you might have to do some changes to make it multiplatform ready.

**Reply by lemkoleg** [reply]
Ok, I'll wait.
